### DISCUSSION TITLE
S3000-G2 as second inverter on same Modbus line

### DISCUSSION THREAD

**Discussion by evaporatingtime** [nathanmarlor/foxess_modbus#668] [Ideas]
Hi all,

I would like to be able to read data from more than one inverter on a single Modbus line.

Having has a quick poke around in the source, it looks to me like the `slave` address does get passes around as a parameter a lot of the time along side the Modbus `client`; this appears encouraging to me at face value, but I'm not that familiar with the project and I develop mostly in c++.

I'd happily give it a go, but could someone help me out with a high-level description of the re-factoring required and/or potential pitfalls?

(As a side note, my second inverter is an S-series G2, so I would also need to add register mapping for this. Though this part seems like it will be much simpler.)

Thanks!

**Comment by canton7** [comment]
Running two inverters is supported already, either on separate adapters or as different slaves on the same adapter. You can add during initial setup, or by going to Configure after.

I don't think the S series talk modbus though? I might be mistaken, search the discussions.

**Reply by canton7** [reply]
Yeah, see https://github.com/nathanmarlor/foxess_modbus/discussions/470 and #231

**Comment by evaporatingtime** [comment]
Thanks for the info and quick reply!

> Running two inverters is supported already, either on separate adapters or as different slaves on the same adapter. You can add during initial setup, or by going to Configure after.

As in, just go through the configuration with the same serial device and a different slave address? I assumed this would overwrite my existing configuration, but I guess I'll try it after confirming the S-G2 will talk.

W.r.t. the S-series G2 (in this case it's a S3000-G2). They've definitely got pins for the RS485 that weren't there for gen 1, and there's a prior illustration of a PC connected.
![image](https://github.com/user-attachments/assets/50ac07ea-0534-4345-818b-4b85704da4b1)


I'm awaiting delivery of some more cabling before I can connect to the S-Series, so I'll grab (yet another!) RS485 interface with the view to identifying the relevant registers in isolation.

**Reply by canton7** [reply]
When you finish the wizard to set up an inverter, you get the option to add another. I think you also get the option to add a second when clicking Configure? It's been a while since I wrote that... Yep if you add the same device with a different slave address, that works as you'd expect.

If you can get it to talk modbus I'll happily add support! Iirc the other s-series also had a pic with a computer attached, but foxess eventually confirmed that it wouldn't talk anything sensible over serial.
